src/lib/openjp2/jp2.c

```diff
@@ -1108,7 +1108,10 @@ static OPJ_BOOL opj_jp2_apply_pclr(opj_image_t *image,
         pcol = cmap[i].pcol;
         src = old_comps[cmp].data;
         assert(src); /* verified above */
-        max = new_comps[pcol].w * new_comps[pcol].h;
+        oldmax = old_comps[cmp].w * old_comps[cmp].h;
```

Comment:
shouldn't the fix to be just:
``max = new_comps[i].w * new_comps[i].h;``

Reply:
It can be... While I understand where and why it accesses out of bands memory, my assumption about the fix can be wrong because I'm not so familiar with the code base. I have applied and tested your suggestion and the crash is gone.